src/EmailArrayField/EmailArrayField.jsx

```diff
@@ -0,0 +1,75 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import ChipArrayField from '../ChipArrayField'
+
+const isValidEmail = (email) => {
+  const rx = /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
+  return rx.test(email)
+}
+
+/**
+ * The `<EmailArrayField>` component allows the editing and validation of a
+ * collection of email addresses.
+ *
+ * It is designed to better experience when entering email addresses by
+ * preventing the need to comma separate the addresses manually.
+ *
+ * ~~~js
+ * import { EmailArrayField } from '@theconversation/ui'
+ *
+ * <EmailArrayField addresses={["hello@unsw.edu.au"]} />
+ * ~~~
+ */
+const EmailArrayField = (props) => <ChipArrayField {...props} onValidateEntry={isValidEmail} />
+
+EmailArrayField.propTypes = {
+  /**
+   * List of entries with which to initialize the component.
+   * Note: This is not a controlled input, it's only read in componentDidMount
+   */
+  entries: PropTypes.array,
+
+  /**
+   * Whether the array should be sorted after each addition
+   */
+  sorted: PropTypes.bool,
+
+  /**
+   * The callback function called to sort the entries.
+   */
+  onSortEntry: PropTypes.func,
+
+  /**
+   * The callback function called when an entry is added or removed.
+   */
+  onChange: PropTypes.func,
+
+  /**
+   * The callback function called to validate the entries. If no function is supplied. All entries are valid
+   */
+  onValidateEntry: PropTypes.func,
+
+  /**
+   * A boolean value indicating whether the text field is disabled.
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * The field label.
+   */
+  label: PropTypes.string,
+
+  /**
+   * The text displayed below the text field.
+   */
+  helperText: PropTypes.string
+}
+
+EmailArrayField.defaultProps = {
+  disabled: false,
+  entries: [],
+  sorted: false
+}
+
+export default EmailArrayField
```

Comment:
It seems that `EmailArrayField` is a custom implementation of `ChipArrayField`. This could be a story of the `ChipArrayField` instead of being a component itself.

Reply:
I was trying to keep ChipArrayField free of any email specific logic. I'm open to suggestions though.

Reply:
I think the `EmailArrayField` is a very specific implementation of the `ChipArrayField`. Don't think it needs to be implemented at `@theconversation/ui`.